src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -195,6 +195,18 @@
   <data name="DoNotUseCountWhenAnyCanBeUsedTitle" xml:space="preserve">
     <value>Do not use Count() or LongCount() when Any() can be used</value>
   </data>
+  <data name="PreferConvertToHexStringOverBitConverterTitle" xml:space="preserve">
+    <value>Prefer 'Convert.ToHexString' over 'BitConverter.ToString'</value>
+  </data>
+  <data name="PreferConvertToHexStringOverBitConverterDescription" xml:space="preserve">
+    <value>'Convert.ToHexString' is more efficient and allocation-friendly than 'BitConverter.ToString' in combination with 'string.Replace'.</value>
+  </data>
+  <data name="PreferConvertToHexStringOverBitConverterMessage" xml:space="preserve">
+    <value>Prefer '{0}' over '{1}'</value>
```

Comment:
NIT: All diagnostics would include `.Replace("-", "")` part right?
```suggestion
    <value>Prefer '{0}' over '{1}.Replace("-", "")'</value>
```

Reply:
There can also be a `ToLower` in between `BitConverter.ToString(data)` and `Replace("-", "")`, or replace could use `string.Empty` as second parameter.

I will try to add better message arguments when modifying the analyzer to also handle `Convert.ToHexStringLower`.